library/src/blas_ex/rocblas_gemm_ext2.cpp

```diff
@@ -0,0 +1,335 @@
+    if(!m || !n)
+        return rocblas_status_success;
+
+    // sizes must not be negative
+    if(m < 0 || n < 0 || k < 0)
+        return rocblas_status_invalid_size;
+
+    // pointers must be valid
+    if(!a || !b || !c || !d || !alpha || !beta)
+        return rocblas_status_invalid_pointer;
+
+    auto stride_a    = rocblas_stride(a_col_stride) * a_row_stride;
+    auto stride_b    = rocblas_stride(b_col_stride) * b_row_stride;
+    auto stride_c    = rocblas_stride(c_col_stride) * c_row_stride;
+    auto stride_d    = rocblas_stride(d_col_stride) * d_row_stride;
+    auto batch_count = 1;
+
+    return rocblas_gemm_ext2_template<false>(handle,
+                                             m,
+                                             n,
+                                             k,
+                                             alpha,
+                                             a,
+                                             a_type,
+                                             0,
+                                             a_row_stride,
+                                             a_col_stride,
+                                             stride_a,
+                                             b,
+                                             b_type,
+                                             0,
+                                             b_row_stride,
+                                             b_col_stride,
+                                             stride_b,
+                                             beta,
+                                             c,
+                                             c_type,
+                                             0,
+                                             c_row_stride,
+                                             c_col_stride,
+                                             stride_c,
+                                             d,
+                                             d_type,
+                                             0,
+                                             d_row_stride,
+                                             d_col_stride,
+                                             stride_d,
+                                             batch_count,
+                                             compute_type)
+}
+
+extern "C" rocblas_status rocblas_gemm_ext2(rocblas_handle    handle,
+                                            rocblas_int       m,
+                                            rocblas_int       n,
+                                            rocblas_int       k,
+                                            const void*       alpha,
+                                            const void*       a,
+                                            rocblas_datatype  a_type,
+                                            rocblas_int       a_row_stride,
+                                            rocblas_int       a_col_stride,
```

Comment:
Does this API have another external equivalent, just wondering why we go with parameter name a_col_stride when it is lda in gemm_ex and expect it is the same thing here.

Reply:
`a_row_stride` and `a_col_stride` make more sense as a pair than `a_row_stride` and `ld_a`. 

We're creating a new API for this use-case.

BLIS, for example, does not use the concept of "leading dimensions'; everything is a row or column stride.

Reply:
or rather `row_stride_a` and `col_stride_` vs `row_stride_a` and `ld_a`